src/functionalTest/kotlin/kotlinx/kover/test/functional/framework/checker/Checker.kt

```diff
@@ -0,0 +1,449 @@
+import org.w3c.dom.*
+import java.io.*
+import javax.xml.parsers.*
+import kotlin.test.*
+
+
+internal fun File.checkResult(
+    result: BuildResult,
+    description: String,
+    errorExpected: Boolean,
+    checker: CheckerContext.() -> Unit
+) {
+    try {
+        createCheckerContext(result).also(checker)
+    } catch (e: Exception) {
+        if (!errorExpected) throw e
+    }
+
+    if (errorExpected) {
+        throw AssertionError("Error expected for $description")
+    }
+}
+
+internal fun File.createCheckerContext(result: BuildResult): CheckerContext {
+    return CheckerContextImpl(this, result, ":")
+}
+
+internal abstract class CheckerContextWrapper(private val origin: CheckerContext) : CheckerContext {
+    override val koverVersion: String?
+        get() = origin.koverVersion
+    override val engine: CoverageEngineVariant
+        get() = origin.engine
+    override val language: ScriptLanguage
+        get() = origin.language
+    override val output: String
+        get() = origin.output
+    override val buildScript: String
+        get() = origin.buildScript
+    override val pluginType: KotlinPluginType?
+        get() = origin.pluginType
+    override val defaultBinaryReport: String
+        get() = origin.defaultBinaryReport
+    override fun allProjects(checker: CheckerContext.() -> Unit) {
+        origin.allProjects(checker)
+    }
+
+    override fun subproject(path: String, checker: CheckerContext.() -> Unit) {
+        origin.subproject(path, checker)
+    }
+
+    override fun output(checker: String.() -> Unit) {
+        origin.output(checker)
+    }
+
+    override fun file(name: String, checker: File.() -> Unit) {
+        origin.file(name, checker)
+    }
+
+    override fun xml(filename: String, checker: XmlReportChecker.() -> Unit) {
+        origin.xml(filename, checker)
+    }
+
+    override fun verification(checker: VerifyReportChecker.() -> Unit) {
+        origin.verification(checker)
+    }
+
+    override fun outcome(taskNameOrPath: String, checker: TaskOutcome.() -> Unit) {
+        origin.outcome(taskNameOrPath, checker)
+    }
+
+    override fun checkReports(xmlPath: String, htmlPath: String, mustExist: Boolean) {
+        origin.checkReports(xmlPath, htmlPath, mustExist)
+    }
+
+    override fun checkOutcome(taskName: String, outcome: TaskOutcome) {
+        origin.checkOutcome(taskName, outcome)
+    }
+
+    override fun checkDefaultReports(mustExist: Boolean) {
+        origin.checkDefaultReports(mustExist)
+    }
+
+    override fun checkDefaultMergedReports(mustExist: Boolean) {
+        origin.checkDefaultMergedReports(mustExist)
+    }
+
+    override fun checkDefaultBinaryReport(mustExist: Boolean) {
+        origin.checkDefaultBinaryReport(mustExist)
+    }
+
+}
+
+private class CheckerContextImpl(
+    val rootDir: File,
+    val result: BuildResult,
+    val path: String
+) : CheckerContext {
+    val projectDir: File = rootDir.sub(path.removeSuffix(":").replace(':', '/'));
```

Comment:
It is quite difficult to implement this, because you need to use the Groovy and Kotlin parser - now it is unjustified, because there are no examples that would use such an opportunity.